### ISSUE TITLE
Kiwix icon low resolution

### ISSUE THREAD

**Issue by LAfricain** [kiwix/kiwix-desktop#525]
The kiwix icon on Ubuntu 20.04 installed with the ppa have a bad resolution.
![kiwix](https://user-images.githubusercontent.com/19690477/94808616-ada1dd00-03f1-11eb-9779-e7889f4bbc2c.png)

**Comment by kelson42** [comment]
@legoktm This is something in the deb packaging I guess, you can point to a picture? Could we simply put a higher resolution?

**Comment by legoktm** [comment]
The Debian packaging doesn't do anything special, it just uses the from `resources/icons/kiwix`.

```
-rw-r--r-- root/root      7532 2020-08-05 22:15 ./usr/share/icons/hicolor/128x128/apps/kiwix-desktop.png
-rw-r--r-- root/root      3752 2020-08-05 22:15 ./usr/share/icons/hicolor/24x24/apps/kiwix-desktop.png
-rw-r--r-- root/root     10689 2020-08-05 22:15 ./usr/share/icons/hicolor/256x256/apps/kiwix-desktop.png
-rw-r--r-- root/root      1692 2020-08-05 22:15 ./usr/share/icons/hicolor/32x32/apps/kiwix-desktop.png
-rw-r--r-- root/root      4640 2020-08-05 22:15 ./usr/share/icons/hicolor/48x48/apps/kiwix-desktop.png
```
I'm guessing @LAfricain has an HDPI display, maybe with some scaling enabled?

https://specifications.freedesktop.org/icon-theme-spec/icon-theme-spec-latest.html suggests that we could also provide an SVG, which I'd hope fixes any resolution issues.

**Comment by kelson42** [comment]
@LAfricain Do you still suffer of the problem with latest version od the dev PPA?

**Comment by LAfricain** [comment]
Yes, still low quality;

**Comment by kelson42** [comment]
@asashnov Is that a topic your are familiar with by any chance?

**Comment by kelson42** [comment]
@asashnov Would you be able please to fix this bug pleae?

**Comment by asashnov** [comment]
@kelson42 I'll have a look. Didn't know about PPA, https://www.kiwix.org/en/download/  says nothing about it.

Found with Google: site:kiwix.org PPA

https://wiki.kiwix.org/wiki/Ubuntu_PPA  - I'll reproduce, test, and try to find the cause of the problem.

**Comment by asashnov** [comment]
Not reproduced for me.

Tried with https://www.linuxvmimages.com/images/ubuntu-2004/

```
$ lsb_release -a
Release:	20.04
Codename:	focal

$ gnome-shell --version
GNOME Shell 3.36.1


$ sudo add-apt-repository ppa:kiwixteam/release

$ sudo apt install kiwix

$ dpkg -l kiwix
ii  kiwix          2.0.5~focal  amd64
```

Press Win key, type 'kiwix' in Gnome Activities search query.

Actual result:

![kiwix-icon-in-gnome-shell](https://user-images.githubusercontent.com/5078434/144596842-2768df4d-6921-44ff-8924-e52d3a8b5551.png)

It doesn't seem low resolution to me.

Also, I checked that the icon sizes provided by 'kiwix' packages actually right. They are:

```
sudo apt install graphicsmagick-imagemagick-compat

$ dpkg -L kiwix | grep icon | grep 'svg\|png'
/usr/share/icons/hicolor/128x128/apps/kiwix-desktop.png
/usr/share/icons/hicolor/24x24/apps/kiwix-desktop.png
/usr/share/icons/hicolor/256x256/apps/kiwix-desktop.png
/usr/share/icons/hicolor/32x32/apps/kiwix-desktop.png
/usr/share/icons/hicolor/48x48/apps/kiwix-desktop.png

### strip the output of 'ientify' after DirectClass to the end of line
$ identify `dpkg -L kiwix | grep icon | grep 'svg\|png'` | sed 's/DirectClass.*//g'
/usr/share/icons/hicolor/128x128/apps/kiwix-desktop.png PNG 128x128+0+0 
/usr/share/icons/hicolor/24x24/apps/kiwix-desktop.png[1] PNG 24x24+0+0 
/usr/share/icons/hicolor/256x256/apps/kiwix-desktop.png[2] PNG 256x256+0+0 
/usr/share/icons/hicolor/32x32/apps/kiwix-desktop.png[3] PNG 32x32+0+0 
/usr/share/icons/hicolor/48x48/apps/kiwix-desktop.png[4] PNG 48x48+0+0
```

@LAfricain my suggestion would be to find in your local ~/.icons and ~/.local/share/icons and even in global /usr/share/icons any 'kiwix-desktop' icon file which may be found by Gnome Shell before the icon provided by the package.

```
find ~/.icons -name 'kiwix-desktop*'
find ~/.local/share/icons -name 'kiwix-desktop*'
find /usr/share/icons -name 'kiwix-desktop*'
```

**Comment by asashnov** [comment]
@LAfricain  Another experiment you could do to see if Gtk API can get the high resolution icon, in my example 128 px, and compare it with what you see by Gnome Image Viewer:

Run both in the same time from 2 different Terminals:

```
$ eog /usr/share/icons/hicolor/128x128/apps/kiwix-desktop.png
```

And in another Terminal:
```
$ python3 icon-view.py
Icon sizes: [256, 128, 32]
```
It also prints the available sizes for the icon. Is 128, 256 is available on your machine for 'kiwix-desktop' icon?

Here is the source code of icon-view.py:

```python
#!/usr/bin/env python3
import gi
gi.require_version("Gtk", "3.0")
from gi.repository import Gtk

# ICON_NAME = "gnome-calculator"
ICON_NAME = "kiwix-desktop"

theme = Gtk.IconTheme.get_default()

print("Icon sizes:", theme.get_icon_sizes(ICON_NAME))

img = Gtk.Image()
b = theme.load_icon(ICON_NAME, 128, 0)
img.set_from_pixbuf(b)

wnd = Gtk.Window()
wnd.connect("destroy", Gtk.main_quit)
wnd.add(img)
img.show()
wnd.show()
Gtk.main()
```

![Screenshot from 2021-12-03 19-06-47](https://user-images.githubusercontent.com/5078434/144600079-ad37f4e9-50f1-443e-bc36-b370623e19ee.png)

**Comment by asashnov** [comment]
Another experiment: I put new .svg icon from #539 to icons store:

```
mkdir -p ~/.local/share/icons/hicolor/scalable/apps
cp kiwix-desktop.svg ~/.local/share/icons/hicolor/scalable/apps
update-icon-caches ~/.local/share/icons
```

Now Gnome Shell gets this icon (but ./icon-view.py shows the same resolutions and icon image):

![Screenshot from 2021-12-03 19-20-52](https://user-images.githubusercontent.com/5078434/144602328-ab7873df-01aa-4de6-824c-f4b407cf15ff.png)

**Comment by asashnov** [comment]
@kelson42  If we are going to use .svg icon in future in Ubuntu, we should prepare it white, or maybe orange, or circled in orange like at the bottom of https://www.kiwix.org/en/

Otherwise in Gnome Shell (Ubuntu, Fedora) it would be nearly invisible:

![Screenshot from 2021-12-03 19-33-12](https://user-images.githubusercontent.com/5078434/144603691-d5d04de3-a193-41c8-af8c-39be0be7d5ff.png)

**Comment by kelson42** [comment]
@asashnov We use to put the logo on a white background to solve this problem.

**Comment by LAfricain** [comment]
Hello,
I use Ubuntu Unity, and I have this issue only in the dock. I'll try to do some check but please wait. I'll have more time the next month.

**Comment by kelson42** [comment]
@asashnov Any news on this ticket?

**Comment by asashnov** [comment]
While researching, I put small text into every icon with commands like:

```
convert /usr/share/icons/hicolor/32x32/apps/kiwix-desktop.png -gravity SouthWest -annotate 0 '32' 32.png
sudo mv 32.png /usr/share/icons/hicolor/32x32/apps/kiwix-desktop.png
```

So I found that 256 px icon is used, scaled down to 96 px:

![Screenshot from 2022-01-12 14-09-43](https://user-images.githubusercontent.com/5078434/149086449-a5098e93-bec6-472a-a295-56f0d3b0bdc0.png)

(you may see small '256' text at left bottom).

**Comment by asashnov** [comment]
So probably addint a proper 96 px size icon may also solve the problem.

But in my fix #762 I added white background to .svg and added it into .deb package to available icons.

Another alternative could be an orange icon, using color #FF9933 from htttp://kiwix.org/  

![Screenshot from 2022-01-12 15-01-52](https://user-images.githubusercontent.com/5078434/149088180-b2570719-4685-47d4-af5f-6f5a4a5f703e.png)

That icon looks good in both white and black themes.
